beater/beater.go

```diff
@@ -116,6 +117,9 @@ func NewCreator(args CreatorParams) beat.Creator {
 			if b.Manager != nil && b.Manager.Enabled() {
 				return nil, errors.New("data streams must be enabled when the server is managed")
 			}
+		} else if bt.config.DataStreams.Enabled && !fleetmode.Enabled() {
+			// not supported only available for development purposes
+			bt.logger.Errorf("Started apm-server with data streams enabled but no active fleet management mode was specified")
 		}
```

Comment:
I wonder if we should simply set 

```go
bt.config.DataStreams.Enabled = fleetmode.Enabled()
```

And disallows it to be set through the external configuration and document the following for testing purposes.

```
 -E management.enabled=true -E management.mode=x-pack-fleet
```

Reply:
That might be reasonable - I'm not sure if enabling fleet management mode when not actually running under Elastic Agent / Fleet would pose any problems; if we can then think that would be fine. The main thing is that I don't want to have to run up all the infrastructure just to test data streams.

Reply:
heh, this change actually trips up a system test which checks that no warnings or errors are logged :)
do you have time to look into whether we can remove this config and just rely on the fleet management config? if not I can look into it some time later

Reply:
>  I'm not sure if enabling fleet management mode when not actually running under Elastic Agent / Fleet would pose any problems; if we can then think that would be fine. 

The elastic-agent-client that the fleet manager uses requires its config in protobuf bytes available on stdIn. So that's a dead end.

> do you have time to look into whether we can remove this config and just rely on the fleet management config? if not I can look into it some time later

Will amend later today!